flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java

```diff
@@ -266,6 +268,20 @@ public Builder upsert(boolean enabled) {
       return this;
     }
 
+    /**
+     * Sets the table loader used to refresh the table instance in {@link IcebergStreamWriter}. If
+     * not specified then the default behavior is to not refresh the table, and the initial table
+     * instance initialized is used for the lifetime of the job.
+     *
+     * @param newTableLoader the table loader to use to refresh the table in writer subtasks
+     * @return {@link Builder} to connect the iceberg table.
+     */
+    @Experimental
+    public Builder writeTableLoader(TableLoader newTableLoader) {
```

Comment:
wondering if we need a separate `writeTableLoader`? @bryanck you mentioned that committer also can have the same problem.  Note that `TableLoader` already has a `clone` method for other reasons/problems. it can be cloned.

if we use the same table loader, I guess we need a way to figure out if the loaded table needs to be refreshed. Should we check the loader class type like `instanceOf CacheTableLoader/RefreshingTableLoader` ? this way we can also have one less public API change.

Reply:
or this can be `tableRefreshInternal(Duration refreshInterval)`?

Reply:
I made this change.